cpp/src/Slice/Parser.cpp

```diff
@@ -3051,105 +3108,34 @@ Slice::Container::validateConstant(const string& name, const TypePtr& lhsType, S
-                        break;
-                    }
-                    case Builtin::KindUShort:
-                    {
-                        min = 0;
-                        max = UINT16_MAX;
-                        break;
-                    }
-                    case Builtin::KindInt:
-                    case Builtin::KindVarInt:
-                    {
-                        min = INT32_MIN;
-                        max = INT32_MAX;
-                        break;
-                    }
-                    case Builtin::KindUInt:
-                    case Builtin::KindVarUInt:
-                    {
-                        min = 0;
-                        max = UINT32_MAX;
-                        break;
-                    }
-                    case Builtin::KindLong:
-                    {
-                        min = INT64_MIN;
-                        max = INT64_MAX;
-                        break;
-                    }
-                    case Builtin::KindULong:
-                    {
-                        min = 0;
-                        max = UINT64_MAX;
-                        break;
-                    }
-                    // The first 2 bits are reserved for storing the length, so we only have 62 bits for the value.
-                    case Builtin::KindVarLong:
-                    {
-                        // We lose another bit here for the sign.
-                        min = -(1LL << 61);
-                        max = (1ULL << 61) - 1;
-                        break;
-                    }
-                    case Builtin::KindVarULong:
-                    {
-                        min = 0;
-                        max = (1ULL << 62) - 1;
-                        break;
-                    }
-                    default:
-                    {
-                        throw logic_error("");
+                        throw out_of_range("");
                     }
                 }
-
-                auto val = stoll(value);
-                if(val < min || val > static_cast<int64_t>(max))
-                {
-                    // `stoll` throws this if the value is outside the range of a long long.
-                    // So it makes sense to piggyback on this and then handle errors all in one place.
-                    throw out_of_range("");
-                }
             }
-        }
-        catch(const out_of_range&)
-        {
-            _unit->error("initializer `" + value + "' for " + desc + " `" + name + "' out of range for type " +
-                         b->kindAsString());
-            return false;
+            catch (const out_of_range&)
+            {
+                ostringstream oss;
+                oss.imbue(underscoreSeparatorLocale);
+                oss << "initializer `" << value << "' for " << desc << " " << name << " is outside type "
+                    << b->kindAsString() << "'s range: [" << min << ".." << max << "]";
```

Comment:
I think this error message sounds awkward, especially the last part about the type's range
"initializer `-1' for constant foo is outside type uint's range: [0..8billionish]"

I think something like:
"constant foo's initializer of `-1' " is outside the range of uint: [0..8billionish]"
Would sound more natural.

Reply:
Fixed.